Fix some xenocrystals

BooJ1K opened the pull request:
Спрайты кристаллов теперь соответствую описанию.

Исправлена ошибка, когда при археологических раскопах выпадал кристалл без описания и c именем "item"

У откопанных соулшардов теперь спрайт обычного соулшарда, что решает проблему с исчезновением спрайта в активной стадией.

fix #4615 
затрагивает ##7009

<details>
<summary>Чейнджлог</summary>

```yml
🆑boojik
bugfix: Исправлена ошибка, когда при археологических раскопках выпадал кристалл с названием "item".
tweak: Спрайты кристаллов теперь соответствую описанию.
tweak: Откопанные соулшарды теперь имеют обычный спрайт и в активном состояние спрайт более не пропадает.
/🆑
```

</details>


- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

BooJ1K committed:
Спрайты кристаллов теперь соответствую описанию.

Исправлена ошибка, когда при археологических раскопах выпадал кристалл без описания и именем "item"

У откопанных соулшардов теперь спрайт обычного соулшарда, что решает проблему с исчезновением спрайта в активной стадией.

fix #4615
затрагивает ##7009

Signed-off-by: BooJ1K <47336771+BooJ1K@users.noreply.github.com>

O0oMaximo0O commented on code/modules/xenoarcheaology/finds/find_spawning.dm:
```diff
@@ -330,6 +334,7 @@
 	else
 		item_type = "rough red crystal"
 		new_item.icon_state = "changerock"
+		return new_item
```
Два других кристалла не вернутся иначе.
```suggestion
	return new_item
```

BooJ1K committed:
Apply suggestions from code review

Co-authored-by: O0oMaximo0O <80180448+O0oMaximo0O@users.noreply.github.com>

Tsurupeta commented on code/modules/xenoarcheaology/finds/find_spawning.dm:
```diff
@@ -315,6 +315,10 @@
+		item_type = "soul stone shard"
+		new_item.icon = 'icons/obj/wizard.dmi'
+		new_item.icon_state = "soulstone"
+		return new_item
```
а зачем ретурн?

BooJ1K replied:
Если тут не оставить ретёрн, то объекты ниже буду по какой-то причине наследовать функцию соулшарда.
https://i.gyazo.com/58a1b35a215ab0143f1adf70c583167d.png

O0oMaximo0O replied:
Потому что это он и будет)))
Просто спрайт возьмёт у кристалла.

Tsurupeta commented on code/modules/xenoarcheaology/finds/find_spawning.dm:
```diff
@@ -330,6 +334,7 @@
 	else
 		item_type = "rough red crystal"
 		new_item.icon_state = "changerock"
+	return new_item
```
и тут

O0oMaximo0O replied:
Надо. Он в инитиалайз археологический возвращается.

BooJ1K replied:
А если тут ретёрн не ставить, происходило смешное из исcуя #4615

BooJ1K committed:
Update find_spawning.dm

BooJ1K committed:
Merge branch 'dev' of https://github.com/BooJ1K/OnyxBay into dev

BooJ1K committed:
Merge branch 'ChaoticOnyx:dev' into dev

BooJ1K committed:
Merge branch 'ChaoticOnyx:dev' into dev

Tsurupeta commented on code/modules/xenoarcheaology/finds/find_spawning.dm:
```diff
@@ -315,6 +315,10 @@
 		new_item = new /obj/item/device/soulstone(loc)
+		item_type = "soul stone shard"
+		new_item.icon = 'icons/obj/wizard.dmi'
+		new_item.icon_state = "soulstone"
```
```suggestion
```

Tsurupeta replied:
У создаваемого соулстоуна и так выставлены icon и icon_state, зачем их ещё раз выставлять?

BooJ1K committed:
Apply suggestions from code review

Изначально не было понятно как поступить с соул шардом, оставить шанс ему иметь спрайты: соулщарда или те 3 вариации, что идут далее, но поскольку у них нет спрайта для активной стадии от этого пришлось отказаться. Спасибо, исправил.

Co-authored-by: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>

BooJ1K committed:
Merge branch 'ChaoticOnyx:dev' into dev
